src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHash.Fixer.cs

```diff
@@ -0,0 +1,128 @@
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var computeHashNode = root.FindNode(context.Span, getInnermostNodeForTie: true);
+            var diagnostics = context.Diagnostics[0];
+            var bufferArgNode = root.FindNode(diagnostics.AdditionalLocations[0].SourceSpan);
+
+            if (computeHashNode is null || bufferArgNode is null)
+            {
+                return;
+            }
+            var hashTypeName = diagnostics.Properties[PreferHashDataOverComputeHashAnalyzer.TargetHashTypeName];
+
+            switch (diagnostics.AdditionalLocations.Count)
+            {
+                case 1:
+                    {
+                        //chained method SHA256.Create().ComputeHash(buffer)
+                        var codeActionChain = new ReplaceNodeHashDataCodeAction(context.Document,
+                            hashTypeName,
+                            bufferArgNode,
+                            computeHashNode);
+                        context.RegisterCodeFix(codeActionChain, diagnostics);
+                        return;
+                    }
+                case 2:
+                    {
+                        var nodeToRemove = root.FindNode(diagnostics.AdditionalLocations[1].SourceSpan);
+                        if (nodeToRemove is null)
+                        {
+                            return;
+                        }
+
+                        if (!TryGetCodeAction(context.Document, hashTypeName, bufferArgNode, computeHashNode, nodeToRemove, out HashDataCodeAction? codeAction))
+                        {
+                            return;
+                        }
+
+                        context.RegisterCodeFix(codeAction, diagnostics);
+                        return;
+                    }
+            }
+        }
+
+        protected abstract bool TryGetCodeAction(Document document, string hashTypeName, SyntaxNode bufferArgNode, SyntaxNode computeHashNode, SyntaxNode nodeToRemove,
+            [NotNullWhen(true)] out HashDataCodeAction? codeAction);
+
+        protected abstract class HashDataCodeAction : CodeAction
+        {
+            protected HashDataCodeAction(Document document, string hashTypeName, SyntaxNode bufferArgNode, SyntaxNode computeHashNode)
+            {
+                Document = document;
+                HashTypeName = hashTypeName;
+                BufferArgNode = bufferArgNode;
+                ComputeHashNode = computeHashNode;
+            }
+            public override string Title => MicrosoftNetCoreAnalyzersResources.PreferHashDataOverComputeHashAnalyzerTitle;
```

Comment:
We should not use the same title as the analyzer title, which generally tends to be longer (normally contains cause and effect). Please a different and much more succinct title that describes the code _fix_, say `Use 'HashData'` or `Replace with 'HashData'`